src/EnergyPlus/InputProcessing/InputProcessor.cc

```diff
@@ -542,6 +542,81 @@ bool InputProcessor::getDefaultValue(EnergyPlusData &state, std::string const &o
     return defaultFound;
 }
 
+std::string InputProcessor::getAlphaFieldValue(
+    EnergyPlusData &state, std::string const &objectWord, json const &ep_object, json schema_obj_props, std::string const &fieldName)
+{
+    auto const &schema_field_obj = schema_obj_props[fieldName];
+    if (schema_field_obj.empty()) {
+        ShowFatalError(state, "InputProcessor::fieldValue: Invalid field name = " + fieldName + "for object type = " + objectWord);
+    }
+    bool isDefaulted = false;
+    std::string value;
+    auto it = ep_object.find(fieldName);
+    if (it != ep_object.end()) {
+        auto const &field_value = it.value();
+        if (field_value.is_string()) {
+            auto valuePair = getObjectItemValue(field_value.get<std::string>(), schema_field_obj);
```

Comment:
I don't see in here where uppercase is set. Is getObjectItemValue where a string gets pushed to uppercase? Can retain case be used here so developers don't make a mistake?

Reply:
Never mind.

    std::pair<std::string, bool> InputProcessor::getObjectItemValue(std::string const &field_value, json const &schema_field_obj)
    {
        if (schema_field_obj.find("retaincase") == schema_field_obj.end()) {
            output.first = UtilityRoutines::MakeUPPERCase(output.first);
        }